### ISSUE TITLE
NullPointerException on connection disposal

### ISSUE THREAD

**Issue by SerCeMan** [rsocket/rsocket-java#914]
Hey, folks!

We've noticed the following error periodically when the connections are being disposed. If I'm reading the exception correctly, it seems like the collection stores `null` at some point.
```
java.lang.NullPointerException: null
    at java.lang.Iterable.forEach(Iterable.java:75)
    at io.rsocket.core.RSocketResponder.cleanUpSendingSubscriptions(RSocketResponder.java:274)
    at io.rsocket.core.RSocketResponder.cleanup(RSocketResponder.java:264)
    at io.rsocket.core.RSocketResponder.tryTerminate(RSocketResponder.java:166)
    at io.rsocket.core.RSocketResponder.tryTerminateOnConnectionClose(RSocketResponder.java:159)
    at reactor.core.publisher.LambdaMonoSubscriber.onComplete(LambdaMonoSubscriber.java:132)
    at reactor.core.publisher.MonoProcessor$NextInner.onComplete(MonoProcessor.java:518)
    at reactor.core.publisher.MonoProcessor.onNext(MonoProcessor.java:308)
    at reactor.core.publisher.MonoProcessor.onComplete(MonoProcessor.java:265)
    at io.rsocket.internal.BaseDuplexConnection.dispose(BaseDuplexConnection.java:23)
    at ...
```

## Steps to Reproduce

It seems that the error isn't easily reproducible. We're only seeing very rarely, once or twice a day. 

## Your Environment

* RSocket version(s) used: 1.0.1
* Other relevant libraries versions (eg. `netty`, ...): none
* Platform (eg. JVM version (`javar -version`) or Node version (`node --version`)): 13
* OS and version (eg `uname -a`): ubuntu 20.04

**Comment by OlegDokuka** [comment]
Hey @SerCeMan!

Thank you for reporting that!
I will have a look. 

On the root-cause... Not sure we put null somewhere, it is more likely it is racing between cleanup during disposal and a stream which does selfremoval when it is terminated as well. So it may be that iterator has seen an element but a collection has already been shrunken so once a next is called, the element being observed is null, hence we see NPE.

**Comment by rstoyanchev** [comment]
This should be resolved now.

**Comment by Cesarla** [comment]
Hi @OlegDokuka, @rstoyanchev  can these improvements be back-ported to rsocket 1.1.X?

**Comment by OlegDokuka** [comment]
@Cesarla any reproducer to share?

**Comment by Cesarla** [comment]
It happens while cleaning channel-based communication:
```
java.lang.NullPointerException: null
at java.lang.Iterable.forEach (Iterable.java:75)
at io.rsocket.core.RSocketResponder.cleanUpSendingSubscriptions (RSocketResponder.java:186)
at io.rsocket.core.RSocketResponder.doOnDispose (RSocketResponder.java:174)
at io.rsocket.core.RSocketResponder.tryTerminate (RSocketResponder.java:108)
at io.rsocket.core.RSocketResponder.tryTerminateOnConnectionClose (RSocketResponder.java:101)
at reactor.core.publisher.LambdaMonoSubscriber.onComplete (LambdaMonoSubscriber.java:135)
at reactor.core.publisher.NextProcessor$NextInner.onComplete (NextProcessor.java:403)
at reactor.core.publisher.NextProcessor.tryEmitValue (NextProcessor.java:185)
at reactor.core.publisher.NextProcessor.tryEmitEmpty (NextProcessor.java:137)
at reactor.core.publisher.NextProcessor.onComplete (NextProcessor.java:132)
at io.rsocket.resume.ResumableDuplexConnection.dispose (ResumableDuplexConnection.java:213)
at io.rsocket.resume.ServerRSocketSession.onNext (ServerRSocketSession.java:215)
at io.rsocket.resume.ServerRSocketSession.onNext (ServerRSocketSession.java:40)
at com.XXXX.eventstream.MdcContextLifter.onNext (MdcContextLifter.java:29)
at reactor.core.publisher.MonoDelay$MonoDelayRunnable.run (MonoDelay.java:119)
at reactor.core.scheduler.SchedulerTask.call (SchedulerTask.java:68)
at reactor.core.scheduler.SchedulerTask.call (SchedulerTask.java:28)
at java.util.concurrent.FutureTask.run (FutureTask.java:264)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:304)
at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
at java.lang.Thread.run (Thread.java:834)
```

**Comment by OlegDokuka** [comment]
yeah. It is actually `IntObjectMap` bug that may send unused values so they are null which endups with NPE

**Comment by yschimke** [comment]
Is this still happening?

**Comment by rstoyanchev** [comment]
@Cesarla this should be fixed in 1.1.1 snapshots.
